src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,28 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+
+        let moveArr: Square[] = []; // Array of squares to move to
+        let currSquare: Square = board.findPiece(this);
+
+        if (this.player === Player.WHITE) {
+            // White moves up, row incremented
```

Comment:
This might seem strange at the moment; I think we could remove all the comments in this file!

We'll come to this tomorrow so no problem for now, but comments that restate the code are often redundant, as our code should be clean enough that its meaning is inherent :)